### DISCUSSION TITLE
Tasmota 2 Smartmeters Wemos D1 Mini and Hichi TTL Reader

### DISCUSSION THREAD

**Discussion by mischa-t** [arendst/Tasmota#21373] [Q&A]
Hello,
i'm not getting my Project running.
My intension is to use 2 x Hichi TTL Readers with one Wemos ESP D1 Mini.
Each Hichi is mounted to a iskra ehz-mt681-d4a52-k0.

I am using a Tasmota Version where SML Scripting is active.
https://ottelo.jimdofree.com/stromz%C3%A4hler-auslesen-tasmota/#Downloads

My Problem now, if i would use only one Hichi everthing work (RX to GPIO16 / TX not connected).
If i try to connect a second or third Hichi only the first wirks the addition doent as if the GPIO would not be active.
I Tried GPIO 16, 14, 4 & 5.

Here my Testing Script:
>D
>B
=>sensor53 r 
>M 4
+1,16,s,0,9600,MT681_1
1,=h==================
1,77070100100700ff@1,Verbrauch,W,Power_curr,0
+2,14,s,0,9600,MT681_2
2,=h==================
2,77070100100700ff@1,Verbrauch,W,Power_curr,0
+3,4,s,0,9600,MT681_3
3,=h==================
3,77070100100700ff@1,Verbrauch,W,Power_curr,0
+4,5,s,0,9600,MT681_4
4,=h==================
4,77070100100700ff@1,Verbrauch,W,Power_curr,0
#

Does anyone know what i am doeing wrong?

**Comment by DittelHome** [comment]
Do you get a answer from cmd sensor53 d1 ?

**Reply by mischa-t** [reply]
Hello, I can only see data from gpio5 with sensors53 d4, all others don't show anything for the above scrip.

**Comment by HofeBY** [comment]
Is there a reason not to activating PullupResistor ?
I think, that "_x" is not valid for Meter Name
I use Try GPIO 2,4,12,15 on an ESP32
replace SN by Number of Meter.

D
B
=>sensor53 r
M 4
+1,2,s,1,9600,AMT681
+2,4,s,1,9600,BMT681
+3,12,s,1,9600,CMT681
+4,15,s,1,9600,DMT681
1,=h=AMT681 SN
1,77070100100700ff@1,A aktLeistung,W,Power_curr,0
2,=h=BMT681 SN
2,77070100100700ff@1,B aktLeistung,W,Power_curr,0
3,=h=CMT681 SN
3,77070100100700ff@1,C aktLeistung,W,Power_curr,0
4,=h=DMT681 SN
4,77070100100700ff@1,D aktLeistung,W,Power_curr,0

**Reply by mischa-t** [reply]
Thanks, I will try and give feedback.
PS. +1,2,s,1,9600,AMT681 is the 1 after the s the pull-up resistor?

**Reply by HofeBY** [reply]
yes

>M 4
; Hier werden die Zähler deffiniert, welche als mqtt/http dargestellt werden.
; ESP32 DevKitC V4 von AZ-Delivery.com
;erster Zähler +1; RxD GPIO 2,Protokoll SML s, Pullup ein 1, Baudrate 9600, Bezeichnung AMT681, TxD GPIO 0, Wartezeit*100ms, Hexstring
+1,2,s,1,9600,AMT681,0,10

**Reply by HofeBY** [reply]
@mischa-t 
Have you ever filmed the transmitter unit with your cell phone camera so that the transmitter is positioned on the receiver.

**Reply by mischa-t** [reply]
Hello, here a little update.
I only get one to works GPIO4.
Also with sensor53 d1 / sensor53 d2 / sensor53 d3 / sensor53 d4 active, i only receive info for sensor53 d4 which is GPIO4.

Both sensors work fine, i tryed them vise versa on different locations with GPIO4.

I wanted to short the rx and tx line but i cant find the tx gpio in the scriopt.
